src/System.Management.Automation/engine/parser/ast.cs

```diff
@@ -8981,7 +8984,7 @@ public ReflectionTypeName(Type type)
         /// <summary>
         /// The name of the assembly.
         /// </summary>
-        public string AssemblyName { get { return _type.Assembly.FullName; } }
+        public string? AssemblyName { get { return _type.Assembly.FullName; } }
```

Comment:
I wonder if this will ever be null. I know that it's annotated as such on `Assembly`, but I haven't been able to produce an instance of it.

@daxian-dbw, do you know?

Reply:
`AssemblyName` is public property in public class `System.Management.Automation.Language.ReflectionTypeName`, I don't think can should assume it can never return null.

Reply:
Since we use .Net API without wrapping we should follow .Net.
I suggest to keep this as is and open new tracking issue in PowerShell repo and new issue in .Net Runtime repo.
Also I guess we could suddenly get null if single file packaging is used.